### PR TITLE
review - wip

### PR DESCRIPTION
**Pull request by FabianLars**
Edit 28.03: frontend vs Frontend
Most of the time Frontend is used but there are many frontend usages too and it's not 100% consistent (or at least it doesn't look like it to me)

### CODE REVIEW

File: src/development/inter-process-communication.md [hunk 1]
```diff
@@ -282,7 +283,7 @@ error type that implements `serde::Serialize`. <br> In the following example, we
 use a crate called [`thiserror`] to help create the error type. It allows you to
 turn enums into error types by deriving the `thiserror::Error` trait. You can
 consult its documentation for more details.
-
+<!-- TODO: in the codesnippet below, consider mentioning that the custom serialize implementation is not needed if the from macro isn't used (or if from is implemented manually). Maybe this would make it too complicated -->
```
Comment (JonasKruckenberg) [inline src/development/inter-process-communication.md:299]: Is that actually true? If you use `#[from]` on all fields you can derive serialize?

Comment (FabianLars) [reply src/development/inter-process-communication.md:299]: What i was trying to say is that the custom Serialize impl is only needed _because_ we use the from macro which stores a non-serialize thing in our error enum. If we remove that and instead use custom from implementations (or map_err everywhere) and have only types that impl Serialize in our error enum we could derive Serialize.

Comment (JonasKruckenberg) [reply src/development/inter-process-communication.md:299]: Ahhhh yeah makes sense! But we shouldn't mention that probably 🤔 a custom macro impl or copious use of `map_err` are not as clean as a simple Serialize implementation

Comment (FabianLars) [reply src/development/inter-process-communication.md:299]: > a custom macro impl

I'm talking about what i did for reqwest errors in my example. Either way i think you're right, that's why i added the "too complicated" thingy.

**Pull request closed without merging**